src/http.rs

```diff
@@ -114,11 +135,14 @@ mod tests {
     use super::*;
     use async_trait::async_trait;
     use regex::Regex;
+    use serde_json::json;
+    use std::collections::HashMap;
+    use std::sync::{Arc, RwLock};
     use uuid::Uuid;
 
     #[derive(Clone, Default)]
     struct MockHttpClient {
-        headers: std::collections::HashMap<String, String>,
+        headers: Arc<RwLock<HashMap<String, String>>>,
```

Comment:
mock client stores the headers that it sent. To make it work in async context I wrapped HashMap in Arc<RwLock>. I also considered DashMap but it would require adding an extra library.